clarity/contracts/alex-vault.clar

```diff
@@ -48,13 +43,13 @@
     (asserts! (> pre-bal amount) insufficient-flash-loan-balance-err)
 
     ;; transfer loan to flash-loan-user
-    (unwrap! (contract-call? token transfer amount (as-contract tx-sender) (contract-of flash-loan-user) none) transfer-failed-err)
+    (unwrap! (contract-call? token transfer amount (as-contract tx-sender) tx-sender none) loan-transfer-failed-err)
 
     ;; flash-loan-user executes with loan received
     (unwrap! (contract-call? flash-loan-user execute) user-execute-err)
 
     ;; return the loan + fee
-    (unwrap! (contract-call? token transfer amount-with-fee (contract-of flash-loan-user) (as-contract tx-sender) none) transfer-failed-err) 
+    (unwrap! (contract-call? token transfer amount-with-fee tx-sender (as-contract tx-sender) none) post-loan-transfer-failed-err) 
     (ok true)
```

Comment:
Thinking of it, I think it will be better to return the transferred amount as a tuple as an return value

Reply:
vault now returns uint (= amount-with-fee).